### PR TITLE
Committing changes required to have Azure Functions depend on the centralized Microsoft BOM file for Java libraries

### PR DESCRIPTION
**Pull request by JonathanGiles**
As discussed via email. There should be no need to introduce version dependencies into this POM any longer, instead depending on the com.microsoft.maven parent pom instead. Please feel free to share any concerns, questions, or anything else - I am happy to answer them and ensure you're comfortable before (and if) you accept this pull request.

### CODE REVIEW

**Comment by pragnagopa** [comment]
- Is there a way for the CI to consume the snapshot plugin?
- Can you please add link to the repo for java-8-parent?

**Comment by JonathanGiles** [comment]
Repo is here: https://github.com/Microsoft/maven-java-parent

Apologies about the build break - I tested locally where the snapshots were already installed. I've updated this PR.

**Comment by pragnagopa** [comment]
PR build still failing. Is the snapshot version available on maven central? If not probably maven settings file should include some info about where to find it
Also, I am trying to figure out testing process for pulling the changes in Library to azure-functions-java-worker. Is it a good practice to publish snapshot version to maven central? Or should we test from so sonatype

**Comment by pragnagopa** [comment]
@JonathanGiles - PR did build with the latest changes but still failing. I will look into it. I can build these changes locally.

File: pom.xml [hunk 1]
```diff
@@ -43,25 +46,36 @@
     </developer>
   </developers>
 
+  <repositories> 
+    <repository>
+      <id>maven.snapshots</id>
+      <name>Maven Central Snapshot Repository</name>
+      <url>https://oss.sonatype.org/content/repositories/snapshots/</url>
+      <releases>
```
Comment (pragnagopa) [inline pom.xml]: How to publish to sonatype snapshots? I need to publish this snapshot version of azure-functions-java-library

**Comment by pragnagopa** [comment]
We have a green build!

**Pull request merged**